### ISSUE TITLE
Add a recent assignment deliveries widget in teacher home screen 

### ISSUE THREAD

**Issue by llaske** [llaske/sugarizer-server#367]
Add a widget in teacher dashboard home screen to see recently delivered assignments.

You can take inspiration from the Recent entries Widget:

![image](https://user-images.githubusercontent.com/981841/219853213-a1cfed71-b5bd-4623-aaba-ea872a501ea7.png)

**Comment by iks1** [comment]
I would like to work on this issue , could you please assign it to me !

**Comment by iks1** [comment]
Basically we have to display some kind of table which displays all the records of assignments, classes etc right ?

**Comment by llaske** [comment]
> I would like to work on this issue , could you please assign it to me !

We don't assign issue. You should check for pull requests linked to an issue you are addressing; if there are none, then assume nobody has done anything.

> Basically we have to display some kind of table which displays all the records of assignments, classes etc right ?

Nope just the list of recently delivered assignments.

**Comment by parteekcoder** [comment]
@llaske I am implementing this, I have a doubt that where can I get the all delivered assignments or where are they recorded that which student has submitted a student

**Comment by parteekcoder** [comment]
@llaske is the endpoint for this task exist or I have to make that , as far as I see there is no endpoint for this task

**Comment by NikhilM98** [comment]
@parteekcoder yeah we are going to need some endpoint for this feature. We can either edit an existing endpoint or create a new endpoint.

- Option 1: Replace `/api/v1/assignments/deliveries/:assignmentId` endpoint with `/api/v1/assignments/deliveries/`.
    - Implement filter by assignment using query parameters.
    - Implement sort by timestamp feature.
    - Update the tests and the UI features using this API.

- Option 2: Create a new endpoint to get all deliveries with filters and sorting.
    - Implement API with sorting/filtering support.
    - Add tests and API documentation for this feature.
    - [Drawback] It will contain redundant logic from get deliveries by assignment feature.

I prefer going with Option 1, although it will may cause backwards incompatibility depending on if Sugarizer is using this API (you can survey this). Moreover, some hardcoded strings in Assignments feature like "Delivered" can me improved.

**Comment by parteekcoder** [comment]
yeah thanks @NikhilM98

**Comment by parteekcoder** [comment]
hey @NikhilM98 here delivered should be true as we are requesting for assignments that are complete

**Comment by Vishalk91-4** [comment]
@llaske, is this issue open for me to work onto

**Comment by llaske** [comment]
@Vishalk91-4 sure, let's go.
